tiledb/api/c_api/array/test/unit_capi_array.cc

```diff
@@ -30,78 +30,1345 @@
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_schema_load(ctx.context, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_schema_free(&schema));
+  CHECK(schema == nullptr);
+}
+
+TEST_CASE(
+    "C API: tiledb_array_schema_load_with_config argument validation",
+    "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_config_t* config{};
+  tiledb_array_schema_t* schema{};
+  /*
+   * No "success" section here; too much overhead to set up.
+   */
+  SECTION("null context") {
+    rc = tiledb_array_schema_load_with_config(
+        nullptr, config, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null config") {
+    // Note: a null config is actually valid and will use the context's config.
+    // This test case merely fails without the proper overhead setup.
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, nullptr, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null uri") {
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, config, nullptr, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, config, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_schema_free(&schema));
+  CHECK(schema == nullptr);
+}
+
+TEST_CASE("C API: tiledb_array_alloc argument validation", "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_array_handle_t* array{};
+  SECTION("success") {
+    rc = tiledb_array_alloc(ctx.context, TEST_URI, &array);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+  }
+  SECTION("null context") {
+    rc = tiledb_array_alloc(nullptr, TEST_URI, &array);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null uri") {
+    rc = tiledb_array_alloc(ctx.context, nullptr, &array);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_alloc(ctx.context, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_free(&array));
+  CHECK(array == nullptr);
+}
+
+TEST_CASE("C API: tiledb_array_free argument validation", "[capi][array]") {
+  ordinary_context ctx{};
+  tiledb_array_handle_t* array{};
+  auto rc = tiledb_array_alloc(ctx.context, TEST_URI, &array);
+  REQUIRE(tiledb_status(rc) == TILEDB_OK);
+  SECTION("success") {
+    REQUIRE_NOTHROW(tiledb_array_free(&array));
+    CHECK(array == nullptr);
+  }
+  SECTION("null array") {
+    /*
+     * `tiledb_array_free` is a void function, otherwise we would check
+     * for an error.
+     */
+    REQUIRE_NOTHROW(tiledb_array_free(nullptr));
+  }
+}
+
+TEST_CASE("C API: tiledb_array_create argument validation", "[capi][array]") {
```

Comment:
This test needs to make a temporary local directory to write the array to. Without it there's not proper test isolation and we'll end up with data races causing random CI failures.